shell/browser/electron_browser_client.cc

```diff
@@ -353,6 +353,49 @@ ElectronBrowserClient::~ElectronBrowserClient() {
   g_browser_client = nullptr;
 }
 
+absl::optional<int> ElectronBrowserClient::GetRenderFrameProcessID() {
```

Comment:
This entire function is flawed logically and needs to be rewritten safely or the code that requires it be refactored out.

This code makes the following assumptions which are unsafe:
* There is only one WebContents in BrowserContext
* There is only one RenderFrameHost in a WebContents (the permission call is for `GetPrimaryMainFrame()`)
* `set_notify_context` is called synchronously on the right thread exactly before this method is called every time (this assumption could be proved incorrect at any time without warning)

This code needs to be deleted and we need to determine the RFH / RPH for a given notification request a different way

Reply:
Hello @MarshallOfSound Your assumptions are incorrect - reaching Notifications module(without hacks) can be only in case of available Notification concept.
@set_notify_context is called synchronously on the right thread exactly before this method is called every time (this assumption could be proved incorrect at any time without warning) - each notifications comes synchronously through database

Reply:
![image](https://user-images.githubusercontent.com/89125090/198222350-4d924d59-c586-4128-a3d9-98d50ed6b14b.png)
Think everything should be much easy for understanding

Reply:
Please do not mark my review comments as resolved, this one is not.

I will be clear, my comment above still stands, I don't care what the current call patterns look like.  Those call patterns are **not** API and are **not** guaranteed in perpetuity. As mentioned above it also makes completely invalid assumptions around which frame to target the request at.

This needs to be changed per my original comment in order to land.